demos/python_demos/face_recognition_demo/face_recognition_demo.py

```diff
@@ -0,0 +1,1073 @@
+                raise NotImplementedError(
+                    "Some layers are not supported on the device")
+
+    def deploy_model(self, model, device, max_requests=1):
+        self.check_model_support(model, device)
+        plugin = self.plugins[device]
+        deployed_model = plugin.load(network=model, num_requests=max_requests)
+        return deployed_model
+
+
+def cut_roi(frame, roi):
+    p1 = roi.position.astype(int)
+    p1 = clip(p1, [0, 0], [frame.shape[-1], frame.shape[-2]])
+    p2 = (roi.position + roi.size).astype(int)
+    p2 = clip(p2, [0, 0], [frame.shape[-1], frame.shape[-2]])
+    return np.array(frame[ :, :, p1[1]:p2[1], p1[0]:p2[0] ])
+
+def cut_rois(frame, rois):
+    return [cut_roi(frame, roi) for roi in rois]
+
+
+class Module(object):
+    def __init__(self, model):
+        self.model = model
+        self.device_model = None
+
+        self.max_requests = 0
+        self.active_requests = 0
+
+        self.clear()
+
+    def deploy(self, device, context, queue_size=1):
+        self.context = context
+        self.max_requests = queue_size
+        self.device_model = context.deploy_model(
+            self.model, device, self.max_requests)
+        self.model = None
+
+    def enqueue(self, input):
+        self.clear()
+
+        if self.max_requests <= self.active_requests:
+            log.warning("Processing request rejected - too many requests")
+            return False
+
+        self.device_model.start_async(self.active_requests, input)
+        self.active_requests += 1
+        return True
+
+    def wait(self):
+        if self.active_requests <= 0:
+            return
+
+        self.perf_stats = [None, ] * self.active_requests
+        self.outputs = [None, ] * self.active_requests
+        for i in range(self.active_requests):
+            self.device_model.requests[i].wait()
+            self.outputs[i] = self.device_model.requests[i].outputs
+            self.perf_stats[i] = self.device_model.requests[i].get_perf_counts()
+
+        self.active_requests = 0
+
+    def get_outputs(self):
+        self.wait()
+        return self.outputs
+
+    def get_performance_stats(self):
+        return self.perf_stats
+
+    def clear(self):
+        self.perf_stats = []
+        self.outputs = []
+
+    def _resize(self, frame, target_shape):
+        assert len(frame.shape) == len(target_shape), \
+            "Expected a frame with %s dimensions, but got %s" % \
+            (len(target_shape), len(frame.shape))
+
+        assert frame.shape[0] == 1, "Only batch size 1 is supported"
+        n, c, h, w = target_shape
+
+        input = frame[0]
+        if not np.array_equal(target_shape[-2:], frame.shape[-2:]):
+            input = input.transpose((1, 2, 0)) # to HWC
+            input = cv2.resize(input, (w, h))
+            input = input.transpose((2, 0, 1)) # to CHW
+
+        return input.reshape((n, c, h, w))
+
+
+class FaceDetector(Module):
+    RETAIL_INPUT_SHAPE = [1, 3, 300, 300]
```

Comment:
This hard-coding doesn't look right. I think there's a method called smth like "getInputsInfo()" can we use that?

Reply:
In #23 we discussed this already - this is done for user input checking. Of course, there are other ways exist to obtain input size, but there it is not for that purpose.

Reply:
I don't see how your discussion in #23 explains the need for hard-coded values. Also, I see absolutely no value in the asserts that you're, probably, talking about (further down the code).